### PR TITLE
Module 2: Capstone Project

### PR DESCRIPTION
**Pull request by RitaDee**
> ### Highlights 🥇
> **RS Movies**
> 
> **Learning objectives**
> 
> * Use webpack to bundle JavaScript.
> * Use callbacks and promises.
> * Learn how to use proper ES6 syntax.
> * Use npm as software packages system.
> * Use callbacks and promises.
> * Use ES6 modules to write modular JavaScript.
> * Send and receive data from an API.
> * Use API documentation.
> * Understand and use JSON.
> * Make JavaScript code asynchronous.
> 
> **What We Have Done:**
> 
> * Added linters. The [JavaScript + HTML + CSS linter](https://github.com/microverseinc/linters-config/tree/master/html-css-js).
> * create a git flow.
> * create a webpack
> * used Es6 import.
> * send and receive data from an API.
> * make use of JavaScript code asynchronous.
> * add the count function
> 
> **Followed Rules:**
> 
> * Made sure [no linter errors](https://github.com/microverseinc/linters-config)
> * Followed [Github Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md)
> * Documented [professionaly](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
> * Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
> * Followed our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### CODE REVIEW

File: src/test/commentCounter.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        creation_date: '2021-02-10',
+        username: 'Jane',
+      },
+    ];
+    const span = document.createElement('span');
+    span.innerHTML = data;
+    const result = (data).length;
+    expect(result).toBe(2);
+  });
+});
```
Comment (CollinsTatang) [inline src/test/commentCounter.test.js:24]: You did a great job implementing the different project requirement 
- [ ] Please your counter functions  fail all the unit test. Kindly correct all the test 
checkOut

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @RitaDee and @shaaibu7  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/test/commentCounter.test.js (hunk 1 above)
Comment (RitaDee) [reply src/test/commentCounter.test.js:24]: Thank you for your review, Collins.


 **PASS  src/test/commentCounter.test.js
 PASS  src/test/moviecount.test.js

Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        0.604 s, estimated 1 s
Ran all test suites.
michaels-mbp:RS-Movie-App michaelobinyan$ npm test

> test
> jest

 PASS  src/test/commentCounter.test.js
 PASS  src/test/moviecount.test.js

Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        0.534 s, estimated 1 s
Ran all test suites.**

This is the result I am getting on my terminal. 
Can you please let me know what you what me to do differently?

Thank you.

Comment (CollinsTatang) [reply src/test/commentCounter.test.js:24]: Hello @RitaDee 
Please can we have a zoom call

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there 👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
✔️ App runs smoothly
✔️ You wrote tests for counter functions

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
- [x] Like the previous reviewer stated, I get the error displayed below when I try to run your tests, you can fix this error by installing jsdom as a dev dependency by running this: ```npm install -D jest-environment-jsdom```. If you get an error subsequently, you can copy that error, paste it on google and you would find a solution to it which would allow all your tests to pass.

Screenshot 2023-02-18 at 16 23 30


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

-  A part of the requirements is that you create tests for counter functions that cover all edge cases for instance the scenario in which there are 0 items on the page (counter function should return 0). Since the previous reviewer did not mention it, I am making it optional but I think implementing it would make your project more secure.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/test/commentCounter.test.js (hunk 1 above)
Comment (RitaDee) [reply src/test/commentCounter.test.js:24]: Yes please.
Thank you

Comment (CollinsTatang) [reply src/test/commentCounter.test.js:24]: Ok

**Review by danmainah — APPROVED** [review]
Hi @RitaDee and @shaaibu7 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RitaDee** [comment]
Thank you, @danmainah
